tests/console/ovs_server.pm

```diff
@@ -0,0 +1,157 @@
+    assert_script_run("cp host_1-privkey.pem $dir_private");
+    assert_script_run("ovs-vsctl set Open_vSwitch . other_config:certificate=/etc/keys/host_1-cert.pem other_config:private_key=/etc/keys/host_1-privkey.pem");
+    assert_script_run("ovs-vsctl add-port br-ipsec tun -- set interface tun type=gre options:remote_ip=$client_ip options:remote_cert=/etc/keys/host_2-cert.pem");
+    systemctl 'restart openvswitch-ipsec';
+
+    # Wait for the client host to setup the IPsec tunnel
+    barrier_wait 'ipsec1_done';
+
+    # Check that ESP packets are being sent from this host(server) to the other(client)
+    assert_script_run("cd");
+    assert_script_run "(ping -c 20 192.0.0.2 &>/dev/null &)";
+    assert_script_run("tcpdump -ni any net -c 10 $client_ip > check");
+    assert_script_run("cat check | grep 'IP $server_ip > $client_ip: ESP'");
+    assert_script_run("rm check");
+
+    barrier_wait 'traffic_check_done1';
+
+    assert_script_run("rm -r $dir*");
+    assert_script_run("rm -r $dir_certs*");
+    assert_script_run("rm -r $dir_private*");
+
+    assert_script_run("ovs-vsctl del-br br-ipsec");
+    add_bridge("$id");
+
+    # Setup IPsec tunnel using CA-signed certificate
+    assert_script_run("ovs-pki init");
+    assert_script_run("cd $dir");
+
+    # Generate and sign the certificate request with the CA key
+    assert_script_run("ovs-pki req -u host_1");
+    assert_script_run("ovs-pki -b sign host_1 switch");
+
+    # Wait for the client to send the certificate request and sign it with the CA key
+    barrier_wait 'host2_cert_ready';
+    assert_script_run("ovs-pki -b sign host_2 switch");
+
+    # Copy the client's certificate and CA certificate to client
+    assert_script_run("scp -o StrictHostKeyChecking=no  host_2-cert.pem root\@$client_ip:/etc/keys/host_2-cert.pem");
+    assert_script_run("scp /var/lib/openvswitch/pki/switchca/cacert.pem root\@$client_ip:/etc/keys/cacert.pem");
+    barrier_wait 'cacert_done';
+
+    # Configure IPsec tunnel to use CA-signed certificate
+    assert_script_run("cp /var/lib/openvswitch/pki/switchca/cacert.pem $dir");
+    assert_script_run("cp /var/lib/openvswitch/pki/switchca/cacert.pem $dir_cacerts");
+    assert_script_run("cp host_1-cert.pem $dir_certs");
+    assert_script_run("cp host_1-privkey.pem $dir_private");
+
+    assert_script_run("ovs-vsctl set Open_vSwitch . other_config:certificate=/etc/keys/host_1-cert.pem other_config:private_key=/etc/keys/host_1-privkey.pem other_config:ca_cert=/etc/keys/cacert.pem");
+    assert_script_run("ovs-vsctl add-port br-ipsec tun -- set interface tun type=gre options:remote_ip=$client_ip options:remote_name=host_2");
+    systemctl 'restart openvswitch-ipsec';
+
+    barrier_wait 'ipsec2_done';
+
+    assert_script_run("cd");
+    assert_script_run "(ping -c 20 192.0.0.2 &>/dev/null &)";
+    assert_script_run("tcpdump -ni any net -c 10 $client_ip > check");
+    assert_script_run("cat check | grep 'IP $server_ip > $client_ip: ESP'");
```

Comment:
Just hint, you can also do grep without cat `grep 'IP $server_ip > $client_ip: ESP' check`. I would use file type check.log or .txt as it obvious it's file thus easier to read. Just `check` can be anything, command or function.